main.go

```diff
@@ -0,0 +1,233 @@
+
+	ServiceAnnotationTagKeyPrefix   = "aws-tag-key/"
+	ServiceAnnotationTagValuePrefix = "aws-tag-value/"
+
+	DescribeTagInputMaxNames = 20
+)
+
+func main() {
+	dry := os.Getenv("DRY_RUN")
+
+	logger := log15.Root()
+	logger.Info("Server Initializing")
+	// Dependency Injection and initialization
+	r := chi.NewRouter()
+	prometheusRegistry := prometheus.NewRegistry()
+
+	prometheusRegistry.MustRegister(prometheus.NewProcessCollector(os.Getpid(), ""))
+	prometheusRegistry.MustRegister(prometheus.NewGoCollector())
+
+	sess := session.Must(session.NewSession())
+	elbAPI := elb.New(sess)
+	//if elbAPI != nil {
+	//	logger.Error("Unable to make connection to AWS", "client", 	elbAPI.ClientInfo )
+	//	return
+	//}
+
+	// Setup middleware global and filters
+	r.Use(
+		middleware.Prometheus(prometheusRegistry),
+		middleware.RequestID,
+		middleware.RequestLogger(logger),
+		middleware.Recoverer,
+	)
+
+	r.Get("/", homeHandler)
+	r.Get("/healthz", healthHandler)
+	r.Mount("/debug", middleware.Profiler())
+	r.Mount("/metrics", promhttp.HandlerFor(prometheusRegistry, promhttp.HandlerOpts{}))
+
+	config, err := kubernetesConfig()
+	if err != nil {
+		panic(err)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err)
+	}
+
+	go func() {
+		for {
+			services, err := clientset.CoreV1().Services("").List(v1.ListOptions{})
+			if err != nil {
+				logger.Error("Error retrieving services on cluster", "err", err)
+			} else {
+
+				serviceTagsToApply := map[string]map[string]string{}
+
+				for _, service := range services.Items {
+					if service.Spec.Type == v1.ServiceTypeLoadBalancer {
+
+						tagsToApply := tagsToApplyFromAnnotations(service.Annotations)
+						if len(tagsToApply) != 0 {
+							// Get the ingress endpoints then tag the associated ELB accordingly
+							for _, ingress := range service.Status.LoadBalancer.Ingress {
+								loadbalancerHostname, err := loadBalancerNameFromHostname(ingress.Hostname)
+								if err != nil {
+									logger.Error("Error parsing the loadbalancer Hostname", "err", err)
+								} else {
+									serviceTagsToApply[loadbalancerHostname] = tagsToApply
+								}
+							}
+						}
+					}
+				}
+
+				logger.Info(fmt.Sprintf("%d Elbs to manage", len(serviceTagsToApply)))
+
+				// TODO: Ideally we should only change tags on elb which needs new tag, to do that we should query
+				// the elb tags list before hand
+				for elbName, tags := range serviceTagsToApply {
+					// FIXME: This we can do in parallel as long as we dont get throttled
+					logger.Info("Applying tag to elb", "elb", elbName, "tags", tags)
+
+					awsTags := []*elb.Tag{}
+					for k, v := range tags {
+						awsTags = append(awsTags, &elb.Tag{
+							Key:   &k,
+							Value: &v,
+						})
+					}
+
+					addTagInput := &elb.AddTagsInput{
+						LoadBalancerNames: []*string{&elbName},
+						Tags:              awsTags,
+					}
+					if strings.ToLower(dry) == "true" {
```

Comment:
I would put this once for all at dry definition:
`dry := strings.ToLower(os.Getenv("DRY_RUN")) == "true"` then use `if dry`